Sage/carbon
# test(badge): refactor badge component

### Proposed behaviour

Refactor for Badge component to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist




#### QA


### Testing instructions

- Run npx cypress open-ct to check if there is newly added test.file
 - Check if the badge.test.js file passed
 - Compare all tests were moved/refactored from .feature file -> .test.js file
 - Check if all unnecessary files were deleted, including .json file
